incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -15,6 +15,32 @@ import (
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 )
 
+func DoSelectorsMatch(inst *unstructured.Unstructured, nsLabels labels.Set) (bool, error) {
+	selector, err := GetSelector(inst)
+	if err != nil {
+		return false, err
+	}
```

Comment:
Nit: personally I'd group them by selector type, so you don't have to "remember" what all the variables are until halfway down the function, e.g.:

```
if sel, err := GetSelector(inst); err != nil {
  return false, err
} else if sel != nil && !sel.Matches(nsLabels) {
  return false, nil
}
// ... same for tree selector ...
if none, err := GetNoneSelector(inst); err != nil || none {
  return false, err
}

```

Note that this doesn't even introduce any temporary variables into the function scope.

Reply:
Ah I see, updated